src/js/server/watcher.js

```diff
@@ -0,0 +1,102 @@
+/*
+
+    A "watcher" that looks for changes in one or more directories and fires an `onFsChange` event if:
+
+    1. Existing files change.
+    2. Files are added.
+    3. Files are removed.
+    4. Files are renamed.
+
+    Based on chokidar: https://github.com/paulmillr/chokidar
+
+ */
+// TODO: Discuss (with Antranig) moving this to its own micro-module if it's more broadly useful.
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+var chokidar = require("chokidar");
+
+fluid.registerNamespace("gpii.handlebars.watcher");
+
+/*
+
+    Initialize our chokidar "watcher" and bind to its events.
+
+ */
+gpii.handlebars.watcher.init = function (that) {
+    var resolvedDirs = fluid.transform(fluid.makeArray(that.options.watchDirs), fluid.module.resolvePath);
+    that.watcher = chokidar.watch(resolvedDirs, that.options.chokidarOptions);
+
+    fluid.each(that.options.eventsToWatch, function (eventName) {
+        // Pass the event type (chokidar only does this for the "all" event type.
+        that.watcher.on(eventName, function (path, details) {
+            that.handleFsEvent(eventName, path, details);
+        });
+    });
+
+    // fluid.log("Watching the following directories: ", resolvedDirs.join(","));
```

Comment:
A few stray commented-out logs in this file - also missing parenthesis on line 33